### ISSUE TITLE
Model Complexity of the Pearson Correlation and LassoCV Algorithms

### ISSUE THREAD

**Issue by EngineerDanny** [EngineerDanny/CS685-Microbe-Network-Research#5]
The goal is to run computational experiments to demonstrate how the `Pearson Correlation Constant` hyper-parameter can be tuned to regularize the `Pearson Correlation Model`. 

The experiment results in a plot of subtrain/validation loss (on the y axis) as a function of a `Pearson Correlation Constant` threshold (on the x axis) with the `American Gut 1` as the main dataset.

Links : [Code](https://github.com/EngineerDanny/CS685-Microbe-Network-Research/blob/main/Fall-2022/week-11/my_pearson_model_complexity.py) ,  [Plot script](https://github.com/EngineerDanny/CS685-Microbe-Network-Research/blob/main/Fall-2022/week-11/my_pearson_model_complexity_plot_script.ipynb)

**Comment by EngineerDanny** [comment]
![Pearson Model Complexity](https://raw.githubusercontent.com/EngineerDanny/CS685-Microbe-Network-Research/main/Fall-2022/week-11/my_pearson_model_complexity_graph.png)

**Comment by EngineerDanny** [comment]
As explained in the issue https://github.com/EngineerDanny/CS685-Microbe-Network-Research/issues/6#issue-1460761866
Scaling and transformation of the datasets is very crucial to the data analysis. Even though the above graph looks great, it was not run with transformed dataset.

**Comment by EngineerDanny** [comment]
![amgut2 FoldID 0](https://raw.githubusercontent.com/EngineerDanny/CS685-Microbe-Network-Research/main/Fall-2022/output/model-complexity/Pearson-Correlation/mc_amgut2_FoldID_0.png)

![amgut1](https://raw.githubusercontent.com/EngineerDanny/CS685-Microbe-Network-Research/main/Fall-2022/output/model-complexity/Pearson-Correlation/amgut_1_FoldID_0.png)

It can inferred from the graph above that Pearson constants closer to zero have the best test errors.

**Comment by EngineerDanny** [comment]
![amgut1_lassocv_FoldID_0](https://raw.githubusercontent.com/EngineerDanny/CS685-Microbe-Network-Research/main/Fall-2022/output/model-complexity/LassoCV/amgut1_lassocv_FoldID_0.png)

![amgut2_lassocv_FoldID_0](https://raw.githubusercontent.com/EngineerDanny/CS685-Microbe-Network-Research/main/Fall-2022/output/model-complexity/LassoCV/amgut2_lassocv_FoldID_0.png)

![baxter_crc_lassocv_FoldID_0](https://raw.githubusercontent.com/EngineerDanny/CS685-Microbe-Network-Research/main/Fall-2022/output/model-complexity/LassoCV/baxter_crc_lassocv_FoldID_0.png)
